package unitarios;

import static org.junit.Assert.*;

import org.junit.AfterClass;
import org.junit.Test;

import dominio.Cliente;
import dominio.ItemPedido;
import dominio.Pedido;
import dominio.Pizza;
import excecoes.ExcecaoDeCliente;
import excecoes.ExcecaoDeItemPedido;
import excecoes.ExcecaoDePedido;
import excecoes.ExcecaoDePizza;

public class ItemPedidoTest {
	
	private static Cliente cliente = null;
	private static Pedido pedido = null;
	private static Pizza pizza = null;
	private static Integer quantidade = null;

	@After
	public static void inicializarDados(){
		try {
			cliente = Cliente.criarCliente("2653 7788", "Jose", "jose@jose.com", "12345", "Rua 10");
		} catch (ExcecaoDeCliente e) {
			e.printStackTrace();
		}
		
		try {
			pedido = Pedido.novoPedido(cliente);
		} catch (ExcecaoDePedido e) {
			e.printStackTrace();
		}
		
		try {
			pizza = Pizza.criarPizza("X-tudo", "Bacon, ovo, presunto, queijo", 25f);
		} catch (ExcecaoDePizza e) {
			e.printStackTrace();
		}
		
		quantidade = 3;
	}
	
	@Test
	public void testCriarItemPedido(){
		try {

			ItemPedido itemPedido = ItemPedido.criarItemPedido(pedido, pizza, quantidade);
			assertNotNull(itemPedido);
			
			assertEquals(pedido, itemPedido.obterPedido());
			assertEquals(pizza, itemPedido.obterPizza());
			assertEquals(quantidade, itemPedido.obterQuantidade());
		} catch (ExcecaoDeItemPedido e) {
			e.printStackTrace();
		}
	}
	
	
}
